### ISSUE TITLE
InstanceNorm eval mode

### ISSUE THREAD

**Issue by darkstar112358** [pytorch/pytorch#1541]
The `InstanceNorm` module in `eval` mode is supposed to behave in the same way as in the `train` mode ([ref](https://github.com/pytorch/pytorch/blob/master/torch/nn/modules/instancenorm.py#L55-L57)), this is done by keeping the module in the `train` mode even if `eval()` is called.

```python
net = torch.nn.InstanceNorm2d(3)
net.eval()
print(net.training) # prints True
```

but if you have `InstanceNorm2d` as a sub-module, the eval mode for `InstanceNorm2d` module doesn't behave as it is expected to:

```python
class CustomNet(torch.nn.Module):
    def __init__(self):
        super(CustomNet, self).__init__()
        self.in1 = torch.nn.InstanceNorm2d(3)

    def forward(X):
        return self.in1(X)

custom_net = CustomNet()
custom_net.eval()
print(custom_net.in1.training) # prints False
```

I suspect this happens because `eval` from a module calls `train(False)`, which bypasses the overload. Is this behavior expected?
